request_handler_proxy.h

```diff
@@ -0,0 +1,39 @@
+#ifndef PROXY_HANDLER_H
+#define PROXY_HANDLER_H
+
+#include "request_handler.h"
+#include "response_parser.h"
+#include <iostream>
+#include <boost/asio.hpp>
+
+namespace http {
+	namespace server {
+		class ProxyHandler : public RequestHandler {
+		public:
+			Status Init(const std::string& uri_prefix, const NginxConfig& config) override;
+
+  			Status HandleRequest(const Request& request, Response* response) override;
+		private:
+			ResponseParser resp_parser;
+
+			// domain name of host to proxy
+			std::string host;
+
+			// port num as a string
+			std::string port;
+
+			// path to server
+			std::string path;
+
+			std::string uri_prefix;
```

Comment:
may want to call the member variables something else or just take out the parameters to the member functions